app/src/main/kotlin/com/hedvig/app/feature/crossselling/ui/CrossSellData.kt

```diff
@@ -50,36 +47,37 @@ data class CrossSellData(
         title = data.title,
         description = data.description,
       )
+
+      fun from(data: CrossSalesQuery.Data.CurrentMember.CrossSell.ProductVariant.Highlight) = Highlight(
+        title = data.title,
+        description = data.description,
+      )
     }
   }
 
   companion object {
-    fun from(data: CrossSellFragment) = CrossSellData(
+    fun from(data: CrossSalesQuery.Data.CurrentMember.CrossSell) = CrossSellData(
+      id = data.id,
       title = data.title,
       description = data.description,
-      callToAction = data.callToAction,
-      action = when {
-        data.action.asCrossSellEmbark?.embarkStoryV2?.name != null -> {
-          val storyId = data.action.asCrossSellEmbark!!.embarkStoryV2.name
-          Action.Embark(storyId, data.title)
-        }
-        data.action.asCrossSellWeb != null -> {
-          val url = data.action.asCrossSellWeb!!.url
-          Action.Web(url)
-        }
-        else -> Action.Chat
-      },
+      about = data.about,
+      storeUrl = data.storeUrl,
       backgroundUrl = data.imageUrl,
       backgroundBlurHash = data.blurHash,
-      crossSellType = data.type.rawValue,
-      typeOfContract = data.contractType.rawValue,
-      about = data.info.aboutSection,
-      perils = data.info.contractPerils.map { Peril.from(it.fragments.perilFragment) },
-      terms = data.info.insuranceTerms.map { DocumentItems.Document.from(it.fragments.insuranceTermFragment) },
-      highlights = data.info.highlights.map(Highlight::from),
-      faq = data.info.faq.map(FAQItem::from),
-      insurableLimits = data.info.insurableLimits.map {
-        InsurableLimitItem.InsurableLimit.from(it.fragments.insurableLimitsFragment)
+      perils = data.productVariants.first().perils.map {
+        Peril.from(it)
+      },
+      terms = data.productVariants.first().documents.map {
+        DocumentItems.Document.from(it)
+      },
+      highlights = data.productVariants.first().highlights.map {
+        Highlight.from(it)
+      },
+      faq = data.productVariants.first().faq.map {
+        FAQItem.from(it)
+      },
+      insurableLimits = data.productVariants.first().insurableLimits.map {
+        InsurableLimitItem.InsurableLimit.from(it)
       },
```

Comment:
If `productVariants` is an empty list (it can be from the GraphQL perspective, nothing preventing it), all the `first()` calls will crash the app.
Should we maybe just do a `firstOrNull()` call and let it be null if the list is empty? 
Also in general, why is `productVariants` a list, but before we were getting a single item? What was the shift that happened here, and do we maybe want to at any point not ignore all the other items in this `productVariants` list?

Reply:
Good catch, fixing